gcc/testsuite/rust/compile/option_env1.rs

```diff
@@ -0,0 +1,21 @@
+#![feature(rustc_attrs)]
+
+#[rustc_builtin_macro]
+macro_rules! option_env {
+    () => {}
+}
+
+#[lang = "sized"]
+trait Sized {}
+
+enum Option<T> {
+    Some(T),
+    None,
+}
```

Comment:
...something like this:

```suggestion
mod core {
mod option {

enum Option<T> {
    Some(T),
    None,
}

}
}
```

Reply:
I fixed this for the tests. However, to get the compilation to work, I needed an extra
`use core::option::Option;`

Is that expected behavior? I would assume so.

Reply:
hmmm... this is weird. you can investigate the generated code by using `-frust-dump-all` and looking at the generated `gccrs.ast-pretty-expanded.dump` file. it should show exactly the code you've generated with your macro expansion

Reply:
This seems like the expected behavior, since we don't have a prelude at the moment:
```rust
liam@gentoo ~/option_env2 $ rustc main.rs
error[E0412]: cannot find type `Option2` in this scope
  --> main.rs:23:13
   |
23 |     let _ : Option2< &'_ str > = core2::option2::Option2::Some("/home/liam/option2_env2");
   |             ^^^^^^^
  --> /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/core/src/option.rs:571:1
   |
   = note: similarly named enum `Option` defined here
   |
help: an enum with a similar name exists
   |
23 |     let _ : Option< &'_ str > = core2::option2::Option2::Some("/home/liam/option2_env2");
   |             ~~~~~~
help: consider importing this enum
   |
9  + use core2::option2::Option2;
   |

error[E0412]: cannot find type `Option2` in this scope
  --> main.rs:24:13
   |
24 |     let _ : Option2< &'_ str > = core2::option2::Option2::None;
   |             ^^^^^^^
  --> /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/core/src/option.rs:571:1
   |
   = note: similarly named enum `Option` defined here
   |
help: an enum with a similar name exists
   |
24 |     let _ : Option< &'_ str > = core2::option2::Option2::None;
   |             ~~~~~~
help: consider importing this enum
   |
9  + use core2::option2::Option2;
   |

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0412`.
liam@gentoo ~/option_env2 $ cat main.rs
//#![feature(rustc_attrs)]
//#[rustc_builtin_macro] macro_rules! option2_env {
//    () => { };
//}

//#[lang = "sized"]
//trait Sized< > { }

pub mod core2 {
    pub mod option2 {
        pub enum Option2< T > {
            Some(T),
            None,
        }

}

}

//use core2::option2::Option2;

fn main() {
    let _ : Option2< &'_ str > = core2::option2::Option2::Some("/home/liam/option2_env2");
    let _ : Option2< &'_ str > = core2::option2::Option2::None;
}
```

Reply:
I did mess up the "pub" on the module names in the test though